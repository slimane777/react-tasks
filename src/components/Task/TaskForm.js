import React from 'react'
import PropTypes from 'prop-types'
import { useSelector, useDispatch } from 'react-redux'
import {
  SET_TASK_TITLE_REQUESTED,
  CREATE_TASK_REQUESTED
} from '../../redux/actions/tasks'
import { InputWrapper, Input, Button } from './styles'
import { FiPlus, FiCheck } from "react-icons/fi";

const TaskForm = () => {

    const title = useSelector((state) => state.title);

    const dispatch = useDispatch();

    const setTaskTitle = (title) => dispatch({ type: SET_TASK_TITLE_REQUESTED, payload: title });
    const createTask = (title) => dispatch({ type: CREATE_TASK_REQUESTED, payload: title });

  const onChange = (e) => {
    setTaskTitle(e.target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()
    createTask(title)
  }

  return (
    
      <form onSubmit={onSubmit}>
        <InputWrapper>
          <FiPlus size={30} color={'#8683a0'} />
          <Input 
            type="text" 
            placeholder="Type your task" 
            onChange={onChange}
            value={title}
          />
          {
            !!title && <Button type="submit">
            <FiCheck size={30} color={'#3dd272'} />
          </Button>
          }
          
        </InputWrapper>
      </form>
  )
}

TaskForm.propTypes = {
  title: PropTypes.string,
//   setTaskTitle: PropTypes.func.isRequired,
//   createTask: PropTypes.func.isRequired
}

export default TaskForm;