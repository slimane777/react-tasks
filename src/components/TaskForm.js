import React from 'react'
import PropTypes from 'prop-types'

// bring connect from react-redux, it's the bridge for connecting component to redux
import { useSelector, useDispatch } from 'react-redux'

// bring the actions, just bring that have REQUESTED in the suffix
// If you dispatching that doesn't have REQUESTED, it will not work
import {
  SET_TASK_TITLE_REQUESTED,
  CREATE_TASK_REQUESTED
} from '../redux/actions/tasks'

const TaskForm = () => {

    const title = useSelector((state) => state.title);
    
    const dispatch = useDispatch();

    const setTaskTitle = (title) => dispatch({ type: SET_TASK_TITLE_REQUESTED, payload: title });
    const createTask = (title) => dispatch({ type: CREATE_TASK_REQUESTED, payload: title });

  const onChange = (e) => {
    setTaskTitle(e.target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()
    createTask(title)
  }

  return (
    <form onSubmit={onSubmit}>
      <input 
	type="text" 
	placeholder="What needs to be done..." 
	onChange={onChange}
	value={title}
      />
      <button type="submit">Submit</button>
    </form>
  )
}

TaskForm.propTypes = {
  title: PropTypes.string,
//   setTaskTitle: PropTypes.func.isRequired,
//   createTask: PropTypes.func.isRequired
}

export default TaskForm;