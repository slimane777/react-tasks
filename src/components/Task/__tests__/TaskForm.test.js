import React from 'react';
import renderer from 'react-test-renderer';
import TaskForm from '../TaskForm';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('TaskForm test', () => {
  const initialState = {tasks:{tasks: []}}
  const mockStore = configureStore()
  let store,wrapper

  it('TaskForm should match snapshot', () => {
    store = mockStore(initialState)
    const component = renderer.create(<Provider store={store}><TaskForm
      title={'title'} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
