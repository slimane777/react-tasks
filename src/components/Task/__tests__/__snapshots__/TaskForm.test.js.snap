// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm test TaskForm should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-bdnxRM kWHvNM"
  >
    <svg
      color="#c0bad2"
      fill="none"
      height={28}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#c0bad2",
        }
      }
      viewBox="0 0 24 24"
      width={28}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="5"
        y2="19"
      />
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
    <input
      className="sc-dlnjwi jxNCaK"
      onChange={[Function]}
      placeholder="Type your task"
      type="text"
    />
  </div>
</form>
`;
